client/src/App/index.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-console */
+import React from 'react';
+import 'antd/dist/antd.css';
+import {
+  BrowserRouter as Router,
+  Route,
+  Switch,
+  Redirect,
+} from 'react-router-dom';
+import ProtectedRoute from './ProtectedRoute';
+import Dashboard from '../pages/Dashboard';
+import City from '../pages/City';
+import Home from '../pages/Home';
+import SignUp from '../pages/SignUp';
+import Login from '../pages/Login';
+import Error from '../pages/Error';
+
+import AuthProvider from '../context/AuthContext';
+
+function App() {
+  return (
+    <AuthProvider>
+      <Router>
+        <Switch>
+          <Route path="/city/:id">
```

Comment:
Whenever you see repeated code, know you're doing something wrong.
Do not repeate code.
Instead of what you're doing in this file, do this:
```jsx
const routes = [
  {path: '/city/:id',
  children: <City />,
  public: true },
 ...
].map(route => route.public ? <Route ...{route} /> : <ProtectedRoute {...route} />)
```